### DISCUSSION TITLE
Local storage

### DISCUSSION THREAD

**Discussion by theNomik** [orestbida/cookieconsent#677] [Q&A] [answered]
Q: Is there any way to overwrite native saving method when 
```
        cookie: {
          useLocalStorage: true,
        },
```
is set? 

Context: Well, when localStorage is used in combination with Capacitor to build iOS/Android app, [storage is getting cleared by OS](https://forum.ionicframework.com/t/strange-bug-ios-local-storage-is-being-erased/147344/7) periodically, so we are forced to use [3rd side storage manager](https://capacitorjs.com/docs/apis/preferences) to save user preferences etc. in async storage.

**Comment by orestbida** [comment, marked as answer]
@theNomik Sorry, this is not possible and there is no plan on adding such option.

With that said, you could still handle this case:

```js
// get value of the the plugin's cookie from your storage manager
const ccValue = storageManager....

// if value is not empty, set it in localStorage
if (value) {
   localStorage.setItem('cc_cookie', JSON.stringify(ccValue));
}

CookieConsent.run({...});
```

You would also need to keep the value updated in your storage manager (update on onAccept and onChange events).

**Reply by theNomik** [reply]
yea, probably most optimal approach, thanks for fast response!
